centralStats/main.bal

```diff
@@ -0,0 +1,173 @@
+        "x-api-key": apiKey
+    };
+    sheets:ConnectionConfig spreadsheetConfig = {
+        auth: {
+            clientId: SPREADSHEET_CLIENT_ID,
+            clientSecret: SPREADSHEET_CLIENT_SECRET,
+            refreshUrl: sheets:REFRESH_URL,
+            refreshToken: SPREADSHEET_REFRESH_TOKEN
+        }
+    };
+
+    sheets:Client spreadsheetClient = check new (spreadsheetConfig);
+
+    // pull count - ballerina/ballerinax
+    string encodedQuery = check url:encode(queryPullCountOfBallerinaBallerinax, "UTF-8");
+    string path = string `/query?query=${encodedQuery}`;
+    HttpResponse response = check http->get(path, headers);
+    _ = check writeDataToSheet(response, spreadsheetClient, SPREADSHEET_ID, "Packages - on Pull packages count - ballerina_ballerinax");
+
+    // pull count - country-wise
+    encodedQuery = check url:encode(queryPullCountByCountry, "UTF-8");
+    path = string `/query?query=${encodedQuery}`;
+    response = check http->get(path, headers);
+    _ = check writeDataToSheet(response, spreadsheetClient, SPREADSHEET_ID, "Country-wise Count - Pull packages count");
+
+    // push count
+    encodedQuery = check url:encode(queryPushCount, "UTF-8");
+    path = string `/query?query=${encodedQuery}`;
+    response = check http->get(path, headers);
+    _ = check writeDataToSheet(response, spreadsheetClient, SPREADSHEET_ID, "Country-wise Count - Push packages count");
+
+    // distribution download count
+    encodedQuery = check url:encode(queryDistDownloadCount, "UTF-8");
+    path = string `/query?query=${encodedQuery}`;
+    response = check http->get(path, headers);
+    _ = check writeDataToSheet(response, spreadsheetClient, SPREADSHEET_ID, "Country-wise Count - Distribution download count");
+
+    // packages - pull count
+    encodedQuery = check url:encode(queryPackages, "UTF-8");
+    path = string `/query?query=${encodedQuery}`;
+    response = check http->get(path, headers);
+    _ = check writeDataToSheet(response, spreadsheetClient, SPREADSHEET_ID, "Packages - on Pull packages count");
+
+}
+
+# Description of the function.
+#
+# + response - Response of the http request
+# + spreadsheetClient - spreadsheetClient configured
+# + spreadsheetID - spreadsheetID of a specific sheet
+# + sheetName - name of the sheet needed to be created
+# + return - Rows to be inserted to the excel sheet
+#
+public function writeDataToSheet(HttpResponse response, sheets:Client spreadsheetClient, string spreadsheetID, string sheetName) returns error? {
+
+    string[] data = [];
+    string[][] excelRows = [];
+    Tables tableResult = <Tables>response.tables[0];
+
+    foreach string[] row in tableResult.rows {
+        data = row;
+        data.push(dateOfQuery);
+        excelRows.push(data);
+    }
+
+    sheets:A1Range a1Range = {
+        sheetName: sheetName
+    };
+    foreach string[] values in excelRows {
+        _ = check spreadsheetClient->appendValue(spreadsheetID, values, a1Range);
+    }
+
+    return ();
```

Comment:
Not required to return nil

Reply:
Fixed https://github.com/ballerina-platform/ballerina-release/pull/2576/commits/dfc15dbb74e88ba2c5a0393162b2316a9ce38169